### PR TITLE
GameDB: Fixes multiple games

### PR DESCRIPTION
**Pull request by RedDevilus**
Fixes (for other games look last comments)
image

### CODE REVIEW

**Comment by lightningterror** [comment]
Maybe add it to other regions too?

**Comment by RedDevilus** [comment]
Only for Socom 3 right?

**Comment by lightningterror** [comment]
Yes

**Comment by prafullpcsx2** [comment]
Are you sure giffifo hack is required for this? Game used to work fine even before that hack was introduced. May be one of the vif hacks is/was better for this game?

**Comment by RedDevilus** [comment]
CK added GIFFIFO to the demo version. Thats how i 'assumed' was the fix. At least its the clamping to 0 one.

**Comment by RedDevilus** [comment]
I shall remove it for now.The other one does work though.

**Comment by RedDevilus** [comment]
Added #2838 to gameDB

**Comment by RedDevilus** [comment]
Going to look for more fixes to the GameDB in the meanwhile don't merge this yet. 😈

**Comment by prafullpcsx2** [comment]
You can add my new patches from this thread:

https://forums.pcsx2.net/Thread-Fixing-unplayable-games

But you will have to first check which were the last patches from that thread which were added to the gamedb. Also skip primal and ghosthunter USA patches as different versions of them have already been added.

**Comment by RedDevilus** [comment]
image
Its odd i dont see the latest patch for theù, and ive based PR on latest master.

**Comment by RedDevilus** [comment]
I have added alot of patches by Prafull and others. Needs checking if i didnt break anything.

**Comment by RedDevilus** [comment]
Overview of the Patches for this PR
SPS:
Primal
SOCOM Combined Assault
SOCOM 3
Amplitude
Next Generation Tennis 2003

Disappearing characters:
SOCOM Combined Assault
SOCOM 3

Hanging:
Gretzky NHL 2005
Speed Kings
Slow performance:
Ghost Hunter
Dengeki D73 demo
NHL 2K6
NHL 2K9

Other (Graphical or runs bad,...):
Savage Skies
True Crime LA (PAL)
UFEA Euro 2004
DT Racer
Free Running
Yanya Caballista
Salt Lake 2002
24 - The Game

Other notes:
Next Generation 2003, Amplitude, Free Running, salt lake, primal, ghosthunter, 24,yanya cabalista are all fixed by rearranging COP2 instructions.

**Comment by RedDevilus** [comment]
Double checked and up for review 👍

File: bin/GameIndex.dbf [hunk 1]
```diff
@@ -1657,6 +1657,17 @@ Serial = SCES-51135
 Name   = Primal
 Region = PAL-M5
 Compat = 5 // With GSdx, SW-mode only.
+vuRoundMode = 0 // Fix SPS
+[patches = dcc4eeea]
+	author=Prafull
+	//Fix slow and choppy gameplay
```
Comment (lightningterror) [inline bin/GameIndex.dbf:1663]: Minor nitpick but comments can be formatted better to suit the rest (ocd satisfaction and consistency).

`//Fix slow and choppy gameplay`
to
`// Fix slow and choppy gameplay.`

Check others too.

Comment (atomic83GitHub) [reply bin/GameIndex.dbf:1663]: Also, it would be nice to add why this patch is require.

Comment (RedDevilus) [reply bin/GameIndex.dbf:1663]: Ive got them from Prafull themselves. I'll ask him.

**Comment by seta-san** [comment]
[Freaky Flyers](https://forums.pcsx2.net/Thread-Freaky-Flyers-SLUS-20284-U--28669) has a simlar slanted shaft coming out of the sky. can't hope it's related?

**Comment by RedDevilus** [comment]
Well its possible its SPS (Spiky Polygon Syndrome). However i haven't done these tests for the games i'm just importing loose patches to main PCSX2. Did you test with the different clamp and rounding methods? If you find the correct settings u can tell me when it does 0 to almost zero bad affects.

**Comment by MrCK1** [comment]
> 
> 
> [Freaky Flyers](https://forums.pcsx2.net/Thread-Freaky-Flyers-SLUS-20284-U--28669) has a simlar slanted shaft coming out of the sky. can't hope it's related?

Freaky Flyers has further issues, the patch in this PR just improves stability running the game.

**Comment by RedDevilus** [comment]
OS got corrupted. I'll do PR again :(

**Pull request closed without merging**